plugins/destination/clickhouse/typeconv/ch/values/decimal.go

```diff
@@ -0,0 +1,26 @@
+package values
+
+import (
+	"math/big"
+
+	"github.com/apache/arrow/go/v12/arrow"
+	"github.com/shopspring/decimal"
+)
+
+type decimalToBigInt interface {
+	BigInt() *big.Int
+}
+
+func decimalValue[A decimalToBigInt](arr primitive[A]) []*decimal.Decimal {
```

Comment:
Same. what the point of generics here? can it be just an interface?

Reply:
Same as Date32/64 but for Decimal128/256